src/pages/javascript/form-validation/index.md

```diff
@@ -2,14 +2,147 @@
 title: Form Validation
 ---
 ## Form Validation
+Form validation normally used to occur at the server, after the client had entered all the necessary data and then pressed the Submit button. If the data entered by a client was incorrect or was simply missing, the server would have to send all the data back to the client and request that the form be resubmitted with correct information. This was really a lengthy process which used to put a lot of burden on the server.
 
-This is a stub. <a href='https://github.com/freecodecamp/guides/tree/master/src/pages/javascript/form-validation/index.md' target='_blank' rel='nofollow'>Help our community expand it</a>.
+JavaScript provides a way to validate form's data on the client's computer before sending it to the web server. Form validation generally performs two functions:
 
-<a href='https://github.com/freecodecamp/guides/blob/master/README.md' target='_blank' rel='nofollow'>This quick style guide will help ensure your pull request gets accepted</a>.
+# Basic Validation - First of all, the form must be checked to make sure all the mandatory fields are filled in. It would require just a loop through each field in the form and check for data.
 
-<!-- The article goes here, in GitHub-flavored Markdown. Feel free to add YouTube videos, images, and CodePen/JSBin embeds  -->
+# Data Format Validation − Secondly, the data that is entered must be checked for correct form and value. Your code must include appropriate logic to test correctness of data.
 
-#### More Information:
-<!-- Please add any articles you think might be helpful to read before writing the article -->
+## Example:
 
+'<html>
```

Comment:
Please make sure you are using the correct code markdown. You should have three backticks (not one) above and below each code segment, like this (I'm using a single quote for example):
```
'''
code
'''
```
Please preview your changes to see the problem.